pkgs/applications/misc/blender/default.nix

```
@@ -225,6 +225,16 @@ stdenv.mkDerivation (finalAttrs: rec {
             --render-frame 1
         done
       '';
+
+      cudaAvailable = runCommand
+      "blender-cuda-available"
+      {
+        nativeBuildInputs = [ finalAttrs.finalPackage ];
+        requiredSystemFeatures = [ "cuda" ];
```

is there a way we could avoid having a stringly-typed API here? eg something like
```suggestion
        requiredSystemFeatures = [ system-features.cuda ];
```
I'm a fan of whatever works, but I think avoiding a stringly-typed interface would be preferable, ceteris paribus.

---

Actually, let me clarify: I made a preset with `"cuda"` because it seemed like the simplest string I could come up with. I'm wary of introducing other names or more infra around these names, because it's easy to slip down the rabbit hole. One obvious rabbit hole is that you could start making derivations that ask for `requiredSystemFeatures = let formatFeature = capabilities: (concatStringsMapSep "-" (x: "sm_{x}") capabilities); [ (formatFeature cudaCapabilities) ]` and deploying hosts that declare `programs.nix-required-mounts.allowedPatterns.cuda.onFeatures = map allowformatFeature (genSubsets cudaCapabilities)`

---

The final value, at least for now, would still be a string, but we could define somewhere a dictionary of valid literals and have people reference them rather than write the strings directly. Could be as silly as `cudaPackages.cudaFeature = "cuda"` and `requiredSystemFeatures = [ cudaPackages.cudaFeature ]`.

We then could even extend this to `cudaFeature = f cudaFlags`

---

But what are your thoughts on modeling the capabilities?

---

I think `(import <nixpkgs> { config.cudaCapabilities = [ "8.6" ]; }).blender.tests.cuda-available.requiredSystemFeatures = [ "cuda-sm_86" ]` would be really great, but we run into the limitations of this mechanism very soon

---

(Nix with an external scheduler when)

---

I think we should model capabilities -- they're essentially features of the hardware which determine how different pieces of software should be built and whether or not they can run on the system.

I do think it's a bit out of scope of this PR though; @SomeoneSerge would you make an issue so we can track this and have a longer discussion there?